tests/test_ingest.py

```diff
@@ -42,25 +42,29 @@ class HscIngestTestCase(IngestTestBase, lsst.utils.tests.TestCase):
     curatedCalibrationDatasetTypes = ("defects", "camera", "bfKernel", "transmission_optics",
                                       "transmission_sensor", "transmission_filter", "transmission_atmosphere")
 
-    def setUp(self):
-        self.ingestDir = os.path.dirname(__file__)
-        self.instrument = "lsst.obs.subaru.HyperSuprimeCam"
-        self.instrumentName = "HSC"
-        self.file = os.path.join(testDataDirectory, "hsc", "raw", "HSCA90402512.fits.gz")
-        self.dataIds = [dict(instrument="HSC", exposure=904024, detector=50)]
-
-        super().setUp()
-        self.butler = Butler(self.root, run=self.outputRun)
-        self.visits = {
+    ingestDir = os.path.dirname(__file__)
+
+    instrumentClassName = "lsst.obs.subaru.HyperSuprimeCam"
+
+    instrumentName = "HSC"
+
+    file = os.path.join(testDataDirectory, "hsc", "raw", "HSCA90402512.fits.gz")
+
+    dataIds = [dict(instrument="HSC", exposure=904024, detector=50)]
+
+    @property
+    def visits(self):
+        butler = Butler(self.root, run=self.outputRun)
```

Comment:
I think this should be:
```python
butler = Butler(self.root, collections=[self.outputRun])
```
because you want it to be read only and you aren't writing any data into it.